dkg/nodesigs.go

```diff
@@ -84,22 +158,25 @@ func (n *nodeSigBcast) exchange(
 		return nil, errors.Wrap(err, "k1 lock hash signature broadcast")
 	}
 
-	ret := make([][]byte, n.operatorAmt)
-	ret[nodeIdx.PeerIdx] = sig
+	n.sigs[n.nodeIdx.PeerIdx] = localSig
```

Comment:
The micro-optimisation to not lock sigs when writing the localsig doesn't feel good. Sure it might be technically correct. But it requires thinking. It requires understanding and remembering that setting the other sigs cannot clash with this sig. This makes the code hard to read and increases cognitive load. Taking the performance hit of always locking and always using `setSig` mitigates that at insignificant cost.